src/runtime_src/core/common/message.cpp

```diff
@@ -277,5 +279,40 @@ send(severity_level l, const char* tag, const char* msg)
     dispatcher->send(l, tag, msg);
   }
 }
+
+void
+send(severity_level l, const char* tag, const char* format, va_list args)
+{
+  static auto verbosity = xrt_core::config::get_verbosity();
+  if (l > (xrt_core::message::severity_level)verbosity) 
+    return;
+  
+  va_list args_bak;
+  // vsnprintf will mutate va_list so back it up
+  va_copy(args_bak, args);
+  int len = std::vsnprintf(nullptr, 0, format, args_bak);
+  va_end(args_bak);
+  if (len <= 0) {
```

Comment:
Is 0 an error?  It's certainly a different condition and not an indication of invalid format string.

Reply:
updated error msg. zero size log msg is unexpected and so reported as error with format string